### PR TITLE
New data format, /tourngame command, lowercase usernames

### LINKED ISSUE #20: Implement a new data storing format

### PR DESCRIPTION
**Pull request by KlicOgogo**
Resolve #20 #18 #19

### CODE REVIEW

File: bot/utils.py [hunk 1]
```diff
@@ -24,10 +21,18 @@ def format_tags(usernames):
 
 
 def parse_game(text):
-    match = re.search(GAME_REGEX, text)
-    if not match:
+    # https://regex101.com/r/9dIo49/2
+    game_regex = r'/game @([^\s]+) (\d+):(\d+) @([^\s]+)'
+    tourngame_regex = r'/tourngame @([^\s]+) (\d+):(\d+) @([^\s]+)'
```
Comment (Drapegnik) [inline bot/utils.py]: 😢

Comment (Drapegnik) [reply bot/utils.py]: https://regex101.com/r/9dIo49/3

File: bot/main.py [hunk 2]
```diff
@@ -104,12 +113,53 @@ def start(message):
 def game(message):
     parsed = parse_game(message.text)
     if not parsed:
-        bot.reply_to(message, f'Invalid format. Use like: `{GAME_FORMAT}`', parse_mode='markdown')
+        game_format = '/game @Drapegnik 5:0 @uladbohdan'
```
Comment (Drapegnik) [inline bot/main.py]:
```
GAME_FORMAT='@drapegnik 5:0 @uladbohdan'
```

```suggestion
        game_format = f'/game {GAME_FORMAT}'
```

File: bot/main.py [hunk 3]
```diff
@@ -104,12 +113,53 @@ def start(message):
...
+    Gameday.games.append(Game(*parsed, None, None))
     bot.send_message(message.chat.id, 'Saved! Use /info')
 
 
+@bot.message_handler(commands=['tourngame'])
+@guard()
+def tournament_game(message):
+    parsed = parse_game(message.text)
+    if not parsed:
+        tourngame_format = '/tourngame @Drapegnik 5:0 @uladbohdan'
```
Comment (Drapegnik) [inline bot/main.py]: same as above

File: bot/main.py [hunk 4]
```diff
@@ -104,12 +113,53 @@ def start(message):
...
+
+    markup = telebot.types.ReplyKeyboardMarkup(row_width=3, one_time_keyboard=True)
+    item_buttons = [telebot.types.KeyboardButton(tournament) for tournament in active_tournaments]
+    markup.add(*item_buttons)
+    msg = bot.send_message(message.chat.id, 'Choose the tournament:', reply_markup=markup)
+    bot.register_next_step_handler(msg, assign_tournament)
+
+
+def assign_tournament(message):
+    print(message.text)
```
Comment (Drapegnik) [inline bot/main.py]:
```suggestion
```

**Review by Drapegnik — CHANGES_REQUESTED** [review]

File: bot/utils.py (hunk 1 above)
Comment (KlicOgogo) [reply bot/utils.py]: 😁

Comment (KlicOgogo) [reply bot/utils.py]: I knew that you will solve this for me. 😊

File: bot/main.py (hunk 2 above)
Comment (KlicOgogo) [reply bot/main.py]: 👍

File: bot/main.py (hunk 4 above)
Comment (KlicOgogo) [reply bot/main.py]: 😔

File: bot/utils.py [hunk 5]
```diff
@@ -21,18 +21,12 @@ def format_tags(usernames):
...
     match_game = re.search(game_regex, text)
-    match_tourngame = re.search(tourngame_regex, text)
-    if not match_game and not match_tourngame:
+    if not match_game:
         return None
-    if match_game:
-        return match_game.groups()
-    else:
-        return match_tourngame.groups()
+    return match_game.groups()[1:]
```
Comment (Drapegnik) [inline bot/utils.py]: to ignore group you can:
`r'/(?:notourngame|game) @([^\s]+) (\d+):(\d+) @([^\s]+)'`

https://www.regular-expressions.info/brackets.html

**Review by Drapegnik — APPROVED** [review]

**Pull request merged**